enterprise/views.py

```diff
@@ -1137,3 +1119,91 @@ def post(self, request, enterprise_uuid, program_uuid):
             )
 
         return redirect(basket_page)
+
+
+class RouterView(NonAtomicView):
+    """
+    A router or gateway view for managing Enterprise workflows.
+    """
+
+    # pylint: disable=invalid-name
+    COURSE_ENROLLMENT_VIEW_URL = '/enterprise/{}/course/{}/enroll/'
+    PROGRAM_ENROLLMENT_VIEW_URL = '/enterprise/{}/program/{}/enroll/'
+    HANDLE_CONSENT_ENROLLMENT_VIEW_URL = '/enterprise/handle_consent_enrollment/{}/course/{}/'
+    VIEWS = {
+        COURSE_ENROLLMENT_VIEW_URL: CourseEnrollmentView,
+        PROGRAM_ENROLLMENT_VIEW_URL: ProgramEnrollmentView,
+        HANDLE_CONSENT_ENROLLMENT_VIEW_URL: HandleConsentEnrollment,
+    }
+
+    @staticmethod
+    def get_path_variables(**kwargs):
+        """
+        Get the base variables for any view to route to.
+
+        Currently gets:
+        - `enterprise_uuid` - the UUID of the enterprise customer.
+        - `course_run_id` - the ID of the course, if applicable.
+        - `program_uuid` - the UUID of the program, if applicable.
+        """
+        enterprise_customer_uuid = kwargs.get('enterprise_uuid', '')
+        course_run_id = kwargs.get('course_id', '')
+        program_uuid = kwargs.get('program_uuid', '')
+        return enterprise_customer_uuid, course_run_id or program_uuid
+
+    def eligible_for_direct_audit_enrollment(self, request, enterprise_customer, course_run_id):
+        """
+        Return whether a request is eligible for direct audit enrollment for a particular enterprise customer.
+
+        We check for the following criteria:
+        - The `audit` query parameter.
+        - The user's being routed to the course enrollment landing page.
+        - The customer's catalog contains the course in question.
+        """
+        # Return it in one big statement to utilize short-circuiting behavior. Avoid the API call if possible.
+        return request.GET.get('audit') and \
+            request.path == self.COURSE_ENROLLMENT_VIEW_URL.format(enterprise_customer.uuid, course_run_id) and \
+            enterprise_customer.catalog_contains_course(course_run_id)
+
+    def redirect(self, request, *args, **kwargs):
+        """
+        Redirects to the appropriate view depending on where the user came from.
+        """
+        enterprise_customer_uuid, resource_id = RouterView.get_path_variables(**kwargs)
+        path = re.sub('{}|{}'.format(enterprise_customer_uuid, re.escape(resource_id)), '{}', request.path)
```

Comment:
This line is a bit opaque to me; can you comment it out?

Reply:
Yes sure. Essentially we're turning the request's path into something that can match one in the class's `VIEWS`, in order to directly and easily match a corresponding view class. Before, we were looping over the dictionary to find a match, but regex is scalable (albeit harder to read, so I'll add that comment).

Reply:
Addressed with 9f7a6fb745c5d3c7abf602745ba170459d1e62ef